lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_generate_strings_file_from_code.rb

```diff
@@ -2,15 +2,32 @@ module Fastlane
   module Actions
     class IosGenerateStringsFileFromCodeAction < Action
       def self.run(params)
-        files = files_matching(paths: params[:paths], exclude: params[:exclude])
-        flags = [('-q' if params[:quiet]), ('-SwiftUI' if params[:swiftui])].compact
-        flags += Array(params[:routines]).flat_map { |routine| ['-s', routine] }
-        cmd = ['genstrings', '-o', params[:output_dir], *flags, *files]
-        out = Actions.sh_control_output(*cmd, print_command: FastlaneCore::Globals.verbose?, print_command_output: true)
-        out = out.scrub.strip.split("\n")
-        errors = out.select { |line| line.include?('genstrings: error: ') }
-        UI.user_error!(errors.join("\n")) unless !params[:fail_on_error] || errors.empty?
-        out
+        output_encoding = begin
+          Encoding.find(params[:output_encoding])
+        rescue ArgumentError => e
+          UI.user_error!(e.message)
+        end
+
+        Dir.mktmpdir('genstrings-output-') do |tmpdir|
+          # Build the command arguments
+          files = files_matching(paths: params[:paths], exclude: params[:exclude])
+          flags = [('-q' if params[:quiet]), ('-SwiftUI' if params[:swiftui]), '-littleEndian'].compact
```

Comment:
Shouldn't `-littleEndian` be conditional as well? 🤔 (e.g. I don't think this is required or if it even works for utf8?)

Reply:
`genstrings` always generates UTF16 output files. The `-littleEndian`/`-bigEndian` optional parameter that you can pass to `genstrings` here is to request `genstrings` to explicitly use a particular endianness for the `UTF-16` output it generates. It is completely unrelated to the `param[:output_encoding]` that the caller requested when calling our action from its `Fastfile`.

If you don't specify `-littleEndian` nor `-bigEndian` when calling `genstrings`, it will still generate output files in UTF16, but using the default endianness of the current OS. Given `genstrings` only runs on Mac, and macOS uses Little Endian, in practice genstrings will always end up generating `UTF-16LE` files even when you don't specify `-littleEndian` explicitly. And it will only generate `UTF-16BE` files if you specify `-bigEndian`, and will never generate `UTF-8` output files.

So in practice that `-littleEndian` was not really strictly necessary (except if one day Apple ports `genstrings` to Linux, and we start being able to run this Fastlane action on Linux machines that use big ending as their default/system endianness…).

But I still preferred to be explicit and provide `-littleEndian` even though `genstrings` run on a Mac will always produce `UTF-16LE` files, **just for consistency** with the fact that, when I later convert the files generated by `genstrings` in `UTF-16` into the `param[:output_encoding]` requested by the Fastlane caller and use `File.read(…, encoding: 'BOM|UTF-16LE')` the encoding that is hardcoded in that `File.read` is guaranteed to match the one endinaness of files produced by `genstrings`.

Reply:
Thanks for the explanation. It wasn't obvious to me.

What do you think of adding a note in the code?

```suggestion
          flags = [
            ('-q' if params[:quiet]),
            ('-SwiftUI' if params[:swiftui]),
            # If no endianness (-bigEndian vs -littleEndian) is specified, genstrings will use endianness of the current OS.
            # Currently, genstrings runs only on macOS, which is little-endian, so this parameter is not strictly necessary.
            # We make it explicit here to raise visibility on the relationship between the endianness of the genstring output and that of the encoding later on.
            '-littleEndian'
          ].compact
```

Reply:
Thanks for the details @AliSoftware -- though hopefully one day we'll see `genstrings` running on Linux 🙏 😅
Nice suggestion @mokagio 👍

Reply:
> hopefully one day we'll see `genstrings` running on Linux 🙏 😅

I mean tbh it's be pretty easy to reimplement `genstrings` in swift using SwiftSyntax nowadays 😅 (see my experiment in https://github.com/AliSoftware/swift-syntax-linter-demo), and then have such a tool run on Linux. The catch being that it wouldn't work on ObjC `.m` sources and only `.swift` (but do we still have that many apps still with ObjC code?), and the tricky part not being the parsing of the swift code after all (thanks to `SyntaxVisitor`'s ease of use) but writing the `.strings` output file correctly in the expected OpenStep format (since everything in Foundation to manipulate plist file formats allow for reading the OpenStep format but not writing it anymore)… so we'd have to write code for that ourselves and be sure to handle edge cases (values with newlines, special characters, …) properly …

So probably not worth it, especially given the future is `.xcstrings` (String Catalogs) anyway (though we'll need GlotPress to support Xliff for that first)… 🤷‍♂️